test/widgets/compose_box_test.dart

```diff
@@ -411,43 +413,52 @@ void main() {
     });
 
     group('topic', () {
-      Future<void> prepareWithTopic(WidgetTester tester, String topic) async {
+      Future<void> prepareWithTopic(WidgetTester tester, String topic, {
+        required int maxTopicLength,
+      }) async {
         TypingNotifier.debugEnable = false;
         addTearDown(TypingNotifier.debugReset);
         MessageStoreImpl.debugOutboxEnable = false;
         addTearDown(MessageStoreImpl.debugReset);
 
         final narrow = ChannelNarrow(channel.streamId);
-        await prepareComposeBox(tester, narrow: narrow, streams: [channel]);
+        await prepareComposeBox(tester, narrow: narrow, streams: [channel],
+          maxTopicLength: maxTopicLength);
         await enterTopic(tester, narrow: narrow, topic: topic);
         await enterContent(tester, 'some content');
       }
 
-      Future<void> checkErrorResponse(WidgetTester tester) async {
+      Future<void> checkErrorResponse(WidgetTester tester, {required int maxTopicLength}) async {
         await tester.tap(find.byWidget(checkErrorDialog(tester,
           expectedTitle: 'Message not sent',
-          expectedMessage: 'Topic length shouldn\'t be greater than 60 characters.')));
+          expectedMessage: 'Topic length shouldn\'t be greater than $maxTopicLength ${maxTopicLength == 1 ? 'character' : 'characters'}.')));
       }
 
+      final variants = ValueVariant({30, 60, 90});
+
       testWidgets('too-long topic is rejected', (tester) async {
-        await prepareWithTopic(tester,
-          makeStringWithCodePoints(kMaxTopicLengthCodePoints + 1));
+        final maxTopicLength = variants.currentValue!;
+        await prepareWithTopic(tester, makeStringWithCodePoints(maxTopicLength + 1),
+          maxTopicLength: maxTopicLength);
```

Comment:
Another consequence of that approach is that these test bodies can avoid getting any more complex than they were: just instead of saying `kMaxTopicLengthCodePoints`, they say `maxTopicLength`, where that's a local constant in this `group` body.